### PR TITLE
Adds ability to snooze Slack reminders for tactical and executive reports

### PR DESCRIPTION
**Pull request by whitdog47**
This adds functionality to the tactical and executive report Slack reminders for the user to be reminded again after a certain time period. Right now, we are only providing a few options (30 minutes, 1 hour, 2 hours) but these are easily modifiable in the `reminder_select_values` dictionary in `messaging/strings.py`. The idea is to test first with a limited set of options to see how users like the feature and what other options they'd like to have. Upon selecting an option, the user either receives a success message or an error with a guid, a new attribute in the incident table stores the reminder time (or null for no reminder), and the scheduler has a new check for the reminder time. The reminder delays are nulled out after a Slack reminder is sent.

### CODE REVIEW

File: src/dispatch/database/service.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def create_sort_spec(model, sort_by, descending):
     """Creates sort_spec."""
     sort_spec = []
     if sort_by and descending:
-        for field, direction in zip(sort_by, descending):
+        for field, direction in zip(sort_by, descending, strict=False):
```
Comment (mvilanova) [inline src/dispatch/database/service.py:410]: I think `strict=False` is the default, isn't it?

File: src/dispatch/decorators.py [hunk 2]
```diff
@@ -52,6 +55,7 @@ def _execute_task_in_project_context(
         )
     except Exception as e:
         # No rollback necessary as we only read from the database
+        log.error("ERROR trying to execute: " + fullname(func))
```
Comment (mvilanova) [inline src/dispatch/decorators.py:58]: f-strings are preferred
```suggestion
        log.error(f"Error trying to execute task: {fullname(func)}")
```

File: src/dispatch/decorators.py [hunk 3]
```diff
@@ -42,6 +42,9 @@ def _execute_task_in_project_context(
                     kwargs["project"] = project
                     func(*args, **kwargs)
             except Exception as e:
+                log.error(
+                    f"ERROR trying to execute database for: {fullname(func)} with parameters {args} and {kwargs}"
```
Comment (mvilanova) [inline src/dispatch/decorators.py:46]:
```suggestion
                    f"Error trying to execute task: {fullname(func)} with parameters {args} and {kwargs}"
```

File: src/dispatch/messaging/strings.py [hunk 4]
```diff
@@ -531,13 +537,41 @@ class MessageType(DispatchEnum):
...
 
+REMIND_AGAIN_OPTIONS = {
+    "text": "[Optional] Remind me again in:",
+    "select": {
+        "placeholder": "Choose a time value",
+        "select_action": ConversationButtonActions.remind_again,
+        "options": [
+            {
+                "option_text": value["message"],
+                "option_value": "{{organization_slug}}-{{incident_id}}-{{report_type}}-" + key,
```
Comment (mvilanova) [inline src/dispatch/messaging/strings.py:548]:
```suggestion
                "option_value": f"{{organization_slug}}-{{incident_id}}-{{report_type}}-key",
```

File: src/dispatch/incident/reminders.py
Comment (mvilanova) [inline src/dispatch/incident/reminders.py:1]: Should this leave in `report/messaging.py`, instead?

File: src/dispatch/messaging/strings.py [hunk 5]
```diff
@@ -531,13 +537,41 @@ class MessageType(DispatchEnum):
...
+            }
+            for key, value in reminder_select_values.items()
+        ],
+    },
+}
+
+
```
Comment (mvilanova) [inline src/dispatch/messaging/strings.py:555]:
```suggestion
```

File: src/dispatch/messaging/strings.py [hunk 6]
```diff
@@ -531,13 +537,41 @@ class MessageType(DispatchEnum):
...
     {
         "title": "{{name}} Incident - {{report_type}} Reminder",
         "title_link": "{{ticket_weblink}}",
         "text": INCIDENT_REPORT_REMINDER_DESCRIPTION,
     },
     INCIDENT_TITLE,
+    REMIND_AGAIN_OPTIONS,
+]
+
+
```
Comment (mvilanova) [inline src/dispatch/messaging/strings.py:566]:
```suggestion
```

File: src/dispatch/plugins/dispatch_slack/incident/enums.py [hunk 7]
```diff
@@ -96,3 +96,7 @@ class UpdateNotificationGroupActionIds(DispatchEnum):
 
 class UpdateNotificationGroupBlockIds(DispatchEnum):
     members = "update-notification-group-members"
+
+
+class RemindAgainActions(DispatchEnum):
+    submit = "remind-again"
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_slack/incident/enums.py:102]:
```suggestion
    submit = ConversationButtonActions.remind_again
```

**Review by mvilanova — CHANGES_REQUESTED** [review]
Nice work so far!

File: src/dispatch/incident/reminders.py
Comment (whitdog47) [reply src/dispatch/incident/reminders.py:1]: Done - thanks, I wasn't sure where to put it. Makes sense.

**Review by mvilanova — APPROVED** [review]

**Pull request merged**